tests/integration_tests/functional/test_api.py

```diff
@@ -724,30 +724,22 @@ def _drive_patch(test_microvm):
 
 def test_api_vsock(test_microvm_with_api):
     """Test vsock related API commands."""
-    if test_microvm_with_api.build_feature != 'vsock':
-        pytest.skip("This test is meant only for vsock builds.")
-
     test_microvm = test_microvm_with_api
     test_microvm.spawn()
     test_microvm.basic_config()
 
     response = test_microvm.vsock.put(
         vsock_id='vsock1',
-        guest_cid=15
-    )
-    assert test_microvm.api_session.is_status_no_content(response.status_code)
-
-    # Adding another vsock should be fine.
-    response = test_microvm.vsock.put(
-        vsock_id='vsock2',
-        guest_cid=16
+        guest_cid=15,
+        uds_path='vsock.sock'
     )
     assert test_microvm.api_session.is_status_no_content(response.status_code)
 
     # Updating an existing vsock is currently fine.
     response = test_microvm.vsock.put(
```

Comment:
Same as above - `uds_path` is present in the update request. If I understand your concern correctly, the Unix socket path should get updated in the same way that the CID does, since the socket is only created at device activation time. Before that, the path is just a string in a config struct.